api/app/routers/fba_calc.py

```diff
@@ -71,39 +74,68 @@ def prepare_response(
         percentage_crown_fraction_burned=fire_behavour_advisory.cfb,
         flame_length=fire_behavour_advisory.flame_length,
         sixty_minute_fire_size=fire_behavour_advisory.sixty_minute_fire_size,
-        thirty_minute_fire_size=fire_behavour_advisory.thirty_minute_fire_size,
-        ffmc_for_hfi_4000=fire_behavour_advisory.ffmc_for_hfi_4000,
-        hfi_when_ffmc_equals_ffmc_for_hfi_4000=fire_behavour_advisory\
-        .hfi_when_ffmc_equals_ffmc_for_hfi_4000,
-        ffmc_for_hfi_10000=fire_behavour_advisory.ffmc_for_hfi_10000,
-        hfi_when_ffmc_equals_ffmc_for_hfi_10000=fire_behavour_advisory\
-        .hfi_when_ffmc_equals_ffmc_for_hfi_10000
+        thirty_minute_fire_size=fire_behavour_advisory.thirty_minute_fire_size
     )
 
     return station_response
 
 
-def process_request(
+async def process_request(
         dailies_by_station_id: dict,
         wfwx_station: WFWXWeatherStation,
         requested_station: StationRequest,
         time_of_interest: datetime,
         date_of_interest: date) -> StationResponse:
     """ Process a valid request """
+    # pylint: disable=too-many-locals
     raw_daily = dailies_by_station_id[wfwx_station.wfwx_id]
 
     # extract variable from wf1 that we need to calculate the fire behaviour advisory.
     bui = raw_daily.get('buildUpIndex', None)
-    ffmc = raw_daily.get('fineFuelMoistureCode', None)
-    isi = raw_daily.get('initialSpreadIndex', None)
-    wind_speed = raw_daily.get('windSpeed', None)
+    temperature = raw_daily.get('temperature', None)
+    relative_humidity = raw_daily.get('relativeHumidity', None)
+    precipitation = raw_daily.get('precipitation', None)
+    # if user has not specified wind speed as part of StationRequest, use the
+    # values retrieved from WFWX in raw_daily
+    if requested_station.wind_speed is None:
+        ffmc = raw_daily.get('fineFuelMoistureCode', None)
+        isi = raw_daily.get('initialSpreadIndex', None)
+        wind_speed = raw_daily.get('windSpeed', None)
+        if raw_daily.get('recordType').get('id') == 'ACTUAL':
+            status = 'Observed'
+        else:
+            status = 'Forecasted'
```

Comment:
There's actual more than Observed and forecast - can you do an explicit check so that it's
`
if raw_daily.get('recordType').get('id') == 'ACTUAL':
            status = 'Observed'
elif raw_daily.get('recordType').get('id') == 'I thing it's FORECAST':
            status = 'Forecasted'
else:
           status = raw_daily.get('recordType').get('id')
`
?
Maybe it's even better just doing:
`
status = raw_daily.get('recordType').get('id')
`

Reply:
Do we really have to match the exact spelling and capitalization? Can't we just use what the wfwx gives us?